app/models/user.rb

```diff
@@ -823,14 +822,11 @@ def check_login
     end
 
     # check if login already exists
-    base_login = login.downcase.strip
+    exists = User.find_by(login: login)
+    return true if !exists || exists.id == id
```

Comment:
On the other hand, maybe we should have a proper check with unique index and `validates_uniqueness_of` validation?

Reply:
Actually, there's already an index for this. Yet no app-side check.